keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -149,22 +149,17 @@ metadata:
   finalizers:
     - cosi.io/finalizer [2]
 spec:
-  protocol:
-    name: [3]
-    version: [4]
-  bucketPrefix: [5]
-  bucketClassName: [6]
-  bucketInstanceName: [7]
+  bucketPrefix: [3]
```

Comment:
Broadly, the 3 fields here seem to represent 2 params to bucket creation and 1 param to bucket attachment (greenfield/brownfield).

How do we think this type will evolve? Given the conversation today, I wonder if this is trying to do too much? I'm not sure, and I know it follows PV/PVC in some ways (and in fact I urged you to follow pattersn), but something feels different.

If we added more fields, how would they likely fall into the green/brown split?

Reply:
> If we added more fields, how would they likely fall into the green/brown split?

In earlier iterations we considered allowing users to specify some bucket attributes in the BR. The same  attributes could also been defined in the BucketClass (BC). Briefly, we defined a `DefaultAttibutes` and `EnforcedAttributes` maps in the BC. The user's BR could override any of the BC's default attributes but none of the enforced attributes.

This ties into the conversation about admin roles vs user role, their overlap, and a need to be able to separate them.  In the end we opted to constrain the user and require the admin to own defining all bucket properties. Generalizing the default vs. enforced idea above, COSI could support the same CRs in both the user and system namespaces. For example, through RBAC the admin could grant users the ability to create their own BCs and thus define bucket properties.

Reply:
I guess what I am getting at is - would more structure here help make clear which fields are associated with each other and which ones are mutually-exclusive?

This sort of (to me) suggests a one-of or even 2 different resources (I will write more on that in the top-level comment)

Reply:
On this note, how (if at all) do you handle *path prefix*?

bucketPrefix seems intended as only for the *bucket name* (e.g., www.googleapis.com/storage/v1/b/<bucketPrefix><bucket.metadata.Name>).

Many "mounting" cases need the ability to have a pathPrefix as well, so:

www.googleapis.com/storage/v1/b/<bucket>/o/<pathPrefix><path>

particularly when used against shared and public buckets.

As @thockin asked, what's the thinking / evolution path here?

Reply:
I've referred this question to @wlan0

Reply:
@thockin > would more structure here help make clear which fields are associated with each other and which ones are mutually-exclusive?

Yes. For MVP we plan on keeping the structure as defined in the KEP, but we are open to improvements in alpha/beta/